airflow-core/docs/installation/upgrading.rst

```diff
@@ -57,22 +57,16 @@ when you choose to upgrade Airflow via their UI.
 How to upgrade
 ==============
 
-Reinstall Apache Airflow®, specifying the desired new version.
-
-To upgrade a bootstrapped local instance, you can set the ``AIRFLOW_VERSION`` environment variable to the
-intended version prior to rerunning the installation command. Upgrade incrementally by patch version: e.g.,
-if upgrading from version 2.8.2 to 2.8.4, upgrade first to 2.8.3. For more detailed guidance, see
-:doc:`/start`.
-
-To upgrade a PyPI package, rerun the ``pip install`` command in your environment using the desired version
-as a constraint. For more detailed guidance, see :doc:`/installation/installing-from-pypi`.
+To upgrade Apache Airflow® installed as PyPI package, rerun the ``pip install`` command in your environment using the desired version set in ``AIRFLOW_VERSION`` environment variable
```

Comment:
I think might be also worth mentioning that a viable option is to create a new virtual environment from the scratch and repeat the installation of airflow following the:
 
* install airflow with constraints (including extras you used initially)
* install additional packages you need  (including extra providers and other packages you did not specify via airflow extras) - without constraints

This is the **right** way of upgrade - because you can easily go back if things go wrong (you still have the old venv) and it requires you to keep the packages you had before and original installation extras somewhere aside.